underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -83,6 +93,9 @@ public int read(byte[] b, int offset, int length) throws IOException {
     int read = mStream.read(b, offset, length);
     if (read != -1) {
       mPos += read;
+      if (mPos % getBlockSize() == 0) {
+        closeStream();
+      }
```

Comment:
Oh also, what will the behavior be if the file is not perfectly block aligned. For example:
```
blockSize = 10
fileLength = 8
```
Will the Swift library give back a 8 length stream or do we need to something like `min(fileLength, endPos)` before putting it in the range?